doc/security/vulnerabilities.rst

```diff
@@ -11,15 +11,380 @@ sufficient time to fix them.  Because the release notes are locked to
+--------------
+
+Multiple Syscalls In GPIO Subsystem Performs No Argument Validation
+
+Multiple syscalls with insufficient argument validation
+
+See NCC-ZEP-006
+
+This has been fixed in releases v1.14.2, and v2.2.0, and in a branch
+from v2.1.0.
+
+- `CVE-2020-10028 <https://docs.zephyrproject.org/latest/security/vulnerabilities.html#CVE-2020-10028>`_
+
+- `Zephyr project bug tracker ZEPSEC-32
+  <https://zephyrprojectsec.atlassian.net/browse/ZEPSEC-32>`_
+
+- `PR23733 fix for v1.14.2
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23733>`_
+
+- `PR23737 fix for branch from v2.1.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23737>`_
+
+- `PR23308 fix for v2.2.0 (gpio patch)
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23308>`_
+
+CVE-2020-10058
+--------------
+
+Multiple Syscalls In kscan Subsystem Performs No Argument Validation
+
+Multiple syscalls in the Kscan subsystem perform insufficient argument
+validation, allowing code executing in userspace to potentially gain
+elevated privileges.
+
+See NCC-ZEP-006
+
+This has been fixed in a branch from v2.1.0, and release v2.2.0.
+
+- `CVE-2020-10058 <https://docs.zephyrproject.org/latest/security/vulnerabilities.html#CVE-2020-10058>`_
+
+- `Zephyr project bug tracker ZEPSEC-34
+  <https://zephyrprojectsec.atlassian.net/browse/ZEPSEC-34>`_
+
+- `PR23748 fix for branch from v2.1.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23748>`_
+
+- `PR23308 fix for v2.2.0 (kscan patch)
+  <https://github.com/zephyrproject-rtos/zephyr/pull/23308>`_
+
+CVE-2020-10059
+--------------
+
+UpdateHub Module Explicitly Disables TLS Verification
+
+The UpdateHub module disables DTLS peer checking, which allows for a
+man in the middle attack. This is mitigated by firmware images
+requiring valid signatures. However, there is no benefit to using DTLS
+without the peer checking.
+
+See NCC-ZEP-018
+
+This has been fixed in a PR against Zephyr master.
+
+- `CVE-2020-10059 <https://docs.zephyrproject.org/latest/security/vulnerabilities.html#CVE-2020-10059>`_
+
+- `Zephyr project bug tracker ZEPSEC-36
+  <https://zephyrprojectsec.atlassian.net/browse/ZEPSEC-36>`_
+
+- `PR24954 fix on master (for v2.3.0)
+  <https://github.com/zephyrproject-rtos/zephyr/pull/24954>`_
+
+- `PR24954 fix v2.1.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/24999>`_
+
+- `PR24954 fix v2.2.0
+  <https://github.com/zephyrproject-rtos/zephyr/pull/24997>`_
+
+CVE-2020-10060
+--------------
+
+UpdateHub Might Dereference An Uninitialized Pointer
+
+In updatehub_probe, right after JSON parsing is complete, objects\[1]
+is accessed from the output structure in two different places. If the
+JSON contained less than two elements, this access would reference
+unitialized stack memory. This could result in a crash, denial of
+service, or possibly an information leak.
+
+Recommend disabling updatehub until such a time as a fix can be made
+available.
+
+See NCC-ZEP-030
+
+This issue has not been fixed.
```

Comment:
Is this necessary ? Can't we just add the link to JIRA ?

Reply:
we already have the link to JIRA though

Reply:
I'd like to be real clear and transparent about these. It goes with why we recommend disabling the feature.